massifs/massifcontextverified.go

```diff
@@ -0,0 +1,221 @@
+package massifs
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+
+	"github.com/datatrails/go-datatrails-common/cose"
+	"github.com/datatrails/go-datatrails-merklelog/mmr"
+)
+
+// Support for an extension to the MassifContext type that provides for getting
+// massif and verifying it at the same time. The primary caller interface is
+// GetVerifiedMassif, and all the other methods are in support of that.  Where
+// possible, and useful, the methods are made available directly on the
+// MassifContext type itself
+
+var (
+	ErrStateSizeBeforeMassifStart = errors.New("the massif index in the mmr state must at least cover the start of the massif")
+	ErrStateSizeExceedsData       = errors.New("There is insufficient data in the massif context to generate a consistency proof against the provided state")
+	ErrSealGetterNotProvided      = errors.New("a seal getter was required but not provided")
+	ErrCBORCodecNotProvided       = errors.New("a CBOR codec was required but not provided")
+	ErrSealNotFound               = errors.New("seal not found")
+	ErrSealVerifyFailed           = errors.New("the seal signature verification failed")
+	ErrGeneratingConsistencyProof = errors.New("error while  creating a consistency proof")
+	ErrConsistencyProofCheck      = errors.New("verification error while checking a consistency proof")
+	ErrInconsistentState          = errors.New("verification failed for a consistency proof")
+	ErrRemoteSealKeyMatchFailed   = errors.New("the provided public key did not match the remote sealing key")
+)
+
+type VerifiedContext struct {
```

Comment:
I think this is an important struct, can we have a struct comment please?

Reply:
Ah, missed that. Yes totally agree